data/modules/scripts/gamestore/init.lua

```diff
@@ -1416,7 +1416,11 @@ function GameStore.processOutfitPurchase(player, offerSexIdTable, addon)
 end
 
 function GameStore.processMountPurchase(player, offerId)
-	player:addMount(offerId)
+if player:hasMount(offerId) then
+		return error({code = 0, message = "You already own this mount."})
+	else
+		player:addMount(offerId)
+	end
```

Comment:
You can correct the indentation and use if with return without needing else.
```suggestion
	if player:hasMount(offerId) then
		return error({code = 0, message = "You already own this mount."})
	end

	player:addMount(offerId)
```